dbms/src/AggregateFunctions/AggregateFunctionAvg.h

```diff
@@ -21,26 +20,25 @@ namespace ErrorCodes
 template <typename T>
 struct AggregateFunctionAvgData
 {
-    T sum = 0;
-    UInt64 count = 0;
+    T numerator = 0;
+    UInt64 denominator = 0;
 
     template <typename ResultT>
     ResultT NO_SANITIZE_UNDEFINED result() const
     {
         if constexpr (std::is_floating_point_v<ResultT>)
             if constexpr (std::numeric_limits<ResultT>::is_iec559)
-                return static_cast<ResultT>(sum) / count; /// allow division by zero
+                return static_cast<ResultT>(numerator) / denominator; /// allow division by zero
 
-        if (count == 0)
+        if (denominator == 0)
             return static_cast<ResultT>(0);
-        return static_cast<ResultT>(sum / count);
+        return static_cast<ResultT>(numerator / denominator);
     }
 };
 
-
 /// Calculates arithmetic mean of numbers.
-template <typename T, typename Data>
-class AggregateFunctionAvg final : public IAggregateFunctionDataHelper<Data, AggregateFunctionAvg<T, Data>>
+template <typename Data, typename T, typename F>
```

Comment:
F is not very good name. maybe rename to `Derived` or something?

Reply:
Why did we change `T` and `Data` order? Don't think it's important, but suspicious.